src/serialization/tokenizer.hpp

```diff
@@ -94,75 +139,78 @@ class tokenizer
 	int lineno_;
 	int startlineno_;
 
+	/**
+	 * increments the line number if the current character is a newline
+	 * set current_ to the next character that's not `\r`
+	 */
 	void next_char()
 	{
-		if (current_ == '\n')
+		if (current_ == token::NEWLINE)
 			++lineno_;
-		next_char_fast();
+		next_char_skip_cr();
 	}
 
-	void next_char_fast()
+	/**
+	 * set current_ to the next character
+	 * skip the `\r` in the `\r\n` Windows-style line endings
+	 * the test_cvs_2018_1999023_2.cfg file also uses `\r\n` line endings for some reason - otherwise that check isn't needed since `\r` characters are removed from cfg files on upload
```

Comment:
It's not true that the check is needed. It's needed for anyone on Windows to play an addon the created themselves.

Reply:
I've clarified the comment.

Reply:
It'd be nice to know if the `\r` is necessary for the test since then the check could be excluded from non-Windows platforms, but that'll be for another PR.